daffodil-runtime1/src/main/scala/org/apache/daffodil/processors/parsers/ZonedTextParsers.scala

```diff
@@ -47,70 +48,69 @@ case class ConvertZonedCombinatorParser(
   }
 }
 
-case class ConvertZonedNumberParser[S](
-  helper: ConvertTextNumberParserUnparserHelperBase[S],
+case class ConvertZonedNumberParser(
   opl: OverpunchLocation.Value,
-  nff: NumberFormatFactoryBase[S],
+  textNumberFormatEv: TextNumberFormatEv,
   zonedSignStyle: TextZonedSignStyle,
   override val context: ElementRuntimeData) extends TextPrimParser {
-  override lazy val runtimeDependencies = Vector()
 
-  override def toString = "to(xs:" + helper.xsdType + ")"
+  override lazy val runtimeDependencies = Vector(textNumberFormatEv)
+
+  private val primNumeric = context.optPrimType.get.asInstanceOf[NodeInfo.PrimType.PrimNumeric]
 
   def parse(start: PState): Unit = {
     val node: DISimple = start.simpleElement
     val str = node.dataValueAsString
 
     Assert.invariant(str != null) // worst case it should be empty string. But not null.
     if (str == "") {
-      PE(start, "Convert to %s (for xs:%s): Cannot parse number from empty string", helper.prettyType, helper.xsdType)
+      PE(start, "Unable to parse zoned %s from empty string", context.optPrimType.get.globalQName)
       return
     }
 
     var checkLength = str.length
     val numValue = {
-      val df = nff.getNumFormat(start)
+      val df = textNumberFormatEv.evaluate(start)
       val pos = new ParsePosition(0)
-      val num = try {
-        val decodedNum = DecimalUtils.zonedToNumber(str, zonedSignStyle, opl)
-        if (decodedNum(0) == '-')
-          checkLength = checkLength + 1
-        df.get.parse(decodedNum, pos)
+
+      val decodedNum = try {
+        DecimalUtils.zonedToNumber(str, zonedSignStyle, opl)
       } catch {
-        case s: scala.util.control.ControlThrowable => throw s
-        case u: UnsuppressableException => throw u
-        case e: Exception => {
```

Comment:
These case never actually cought anything. I think they were copied from the text number parser, which also never caught anything and were removed in this PR. I suspect they could have happened in an older versions of ICU.